golem-worker-service-base/src/gateway_execution/auth_call_back_binding_handler.rs

```diff
@@ -24,14 +23,15 @@ use golem_common::SafeDisplay;
 use openidconnect::core::CoreTokenResponse;
 use openidconnect::{AuthorizationCode, OAuth2TokenResponse};
 use std::sync::Arc;
+use url::Url;
 
 pub type AuthCallBackResult = Result<AuthorisationSuccess, AuthorisationError>;
 
 #[async_trait]
 pub trait AuthCallBackBindingHandler {
     async fn handle_auth_call_back(
         &self,
-        http_request_details: &HttpRequestDetails,
+        api_url: &Url,
```

Comment:
The design I went for is that all input specific (http, grpc, ...) details are handled in the executor. By the time we are reaching the binding handlers, no input specific data should show up